### ISSUE TITLE
Mac client stuck at migration

### ISSUE THREAD

**Issue by jeffdomke** [blockstack/blockstack-browser#1430]
![Screen Shot 2018-05-31 at 3:18 PM](http://marker.screenshots.prod.s3.amazonaws.com/33ae9cf956154becbdf8b2b140405865-1527779936225.png)
---
### More Info

**Source URL**: [http://localhost:8888/update?redirect=%2F%23coreAPIPassword%3D825227B4-9151-4DBB-88EF-815E024F2D4D-71787-00005C7DE1660128%26logServerPort%3D8333%26regtest%3D0](http://localhost:8888/update?redirect=%2F%23coreAPIPassword%3D825227B4-9151-4DBB-88EF-815E024F2D4D-71787-00005C7DE1660128%26logServerPort%3D8333%26regtest%3D0)

**Comment by jeffdomke** [comment]
![screen shot 2018-05-31 at 11 18 36 am](https://user-images.githubusercontent.com/767828/40791021-6de02678-64c4-11e8-80a1-4558110aed76.png)

**Comment by aulneau** [comment]
cc @yknl looks like it's a similar bug to what we had before.

**Comment by jeffdomke** [comment]
Conditions that might have caused, I have the previous browser installed, but may have hit reset prior to installation. I just re-opened the old version after doing the above and it is also now borked.

```
Uncaught TypeError: Cannot read property 'length' of undefined
    at main.js:176855
    at dispatch (main.js:179392)
    at createStore (main.js:179467)
    at main.js:178828
    at main.js:176984
    at main.js:178960
    at configureStore (main.js:16796)
    at Object.<anonymous> (main.js:6225)
    at Object.require.42../routes (main.js:6256)
    at s (main.js:1)
```

**Comment by yknl** [comment]
@jeffdomke by "hit reset" do you mean this? 
![image](https://user-images.githubusercontent.com/29081231/40791280-25036a4a-64c5-11e8-9060-890200907e7b.png)

**Comment by aulneau** [comment]
I think we need to write a function to check the persisted state and see if there is no encrypted backup phrase, and pull from the `redux_old` and reset it to the latest computedState stored in the backup, and then start the updating process over. I can take a look this afternoon on this issue. cc @yknl

**Comment by aulneau** [comment]
but that action might not clear everything we need it to (eg the state version is still persisted which means that the app will try to update). We have to add in some additional logic to deal with this kind of scenario

**Comment by jeffdomke** [comment]
Yeah, I didn't do that on purpose, I was just reseting a ton the last few days and happened to stumble into that edge case. I just tried this on another chrome user, reset everything and reproduced. Resetting the blockstack browser before migrating causes this issue 100%

**Comment by aulneau** [comment]
Yeah totally. I think we need to just blow up the state and redirect to the restore page in this case because there's nothing else we can do (and it would be accomplishing the same thing)
